crates/sui-core/src/authority/authority_store_types.rs

```diff
@@ -12,10 +12,77 @@ use sui_types::error::SuiError;
 use sui_types::move_package::MovePackage;
 use sui_types::object::{Data, MoveObject, Object, Owner};
 
+// Versioning process:
```

Comment:
FWIU with the current approach:
- we'll have to keep old code around for all previous versions
- every read will trigger a chain of migrations until the value gets overwritten. If the value is immutable, it will just stay the same. Such ad hoc migrations can also have a CPU impact
- it won't allow performing more involved migrations like the recent switch from `DBMap<ObjectKey, Object>` to a pair of `DBMap<ObjectKey, StoreObject>` and `DBMap<ObjectContentDigest, StoreMoveObject>`
- adding extra wrapper layers adds a bit of complexity 

I did a quick look at what others are doing. Looks like a traditional migration approach is utilized (https://near-nodes.io/maintenance/db-migrations). Did we consider that?

Reply:
> every read will trigger a chain of migrations until the value gets overwritten

no, only reads of very old keys, which are definitionally rare

> it won't allow performing more involved migrations like the recent switch from

that's true, but it doesn't preclude it either

> Looks like a traditional migration approach is utilized (https://near-nodes.io/maintenance/db-migrations). Did we consider that?

We did - I was worried that eager migrations might be too slow once the db is very large.